### PR TITLE
Feature Request: add checking support and md5 hashing for getModulus

### PR DESCRIPTION
**Pull request by KaiSchwarz-cnic**
as of: https://github.com/Dexus/pem/issues/126

I've implemented the certificate checks and added md5 hashing to getModulus.
As pointed out, i would prefer to do some piping instead of introducing md5.

I've not covered further format conversions as for our purpose the checks are important, further conversion possibilities: nice 2 have.

Best
Kai

### CODE REVIEW

**Comment by KaiSchwarz-cnic** [comment]
I've applied some fixes, as I forgot little things in hurry.

Tests and Documentation are reviewed accordingly.
checkPkcs12 is also working fine, but I don't get why the result is to be find in stderr instead of stdout.
Maybe you can clarify this :)

**Comment by KaiSchwarz-cnic** [comment]
I just have no glue why grunt exits with code 91, maybe you can help here @Dexus?

**Comment by Dexus** [comment]
Will review when i have time. I'm a bit busy until next mondey.

Von meinem iPhone gesendet

File: test/pem.js [hunk 1]
```diff
@@ -625,6 +665,43 @@ exports['General Tests'] = {
...
+    'Check PKCS12 keystore': function(test) {
+      pem.createCertificate({
+          commonName: 'CA Certificate'
+      }, function (error, ca) {
+        test.ifError(error);
+
+        pem.createCertificate({
+            serviceKey: ca.serviceKey,
+            serviceCertificate: ca.certificate,
+            serial: Date.now(),
```
Comment (Dexus) [inline test/pem.js]: I think this makes the Error Code 91.

Serial need to refactored as of #84

**Review by Dexus — CHANGES_REQUESTED** [review]
:LGTM:

File: test/pem.js (hunk 1 above)
Comment (KaiSchwarz-cnic) [reply test/pem.js]: Ok, thanks for checking this. Sorry for asking, but can you give that some priority in the next week(s)? Or am I understanding this wrong: Do I need to change the serial in my code or is there a change on the serial option required on your side?

**Comment by papakai** [comment]
@Dexus I rewrote the checkPkcs12 test, but finally this did not change things. Maybe because other existing tests (not added by myself) are also using the serial. What I do not understand here, is why this could succeed previously in other commits I've made.

**Comment by KaiSchwarz-cnic** [comment]
Found the stupid issue - sorry for that. got fixed and now we're ready to merge :+1:

**Review by Dexus — APPROVED** [review]

**Comment by Dexus** [comment]
Thanks, will release as soon as possible the new version

**Comment by KaiSchwarz-cnic** [comment]
Many thanks to you for merging :+1:

**Comment by KaiSchwarz-cnic** [comment]
@Dexus 
Even though I do not want to pressure...
Any news on the npm version update? I'm just asking, because increasing version number and publishing the update to npmjs.org doesn't take more than 5 minutes :) Or do you need to include further things?

**Comment by Dexus** [comment]
I need access to my home server thats all...

Von meinem iPhone gesendet

**Comment by KaiSchwarz-cnic** [comment]
@Dexus you might also think about adding me as collaborator to npm, so that I might support in future too. Let me know if that would be of interest.

**Comment by Dexus** [comment]
Hi @papakai I have send you an invite: https://github.com/Dexus/pem/invitations

**Comment by KaiSchwarz-cnic** [comment]
@Dexus Thanks, very appreciated. I have already some ideas, but can only start with things when I have the time for it :-) At least we should start with introducing some source code beautifying with a default configuration file. That would allow to reformat code written by the community which is very important imo.

**Comment by Dexus** [comment]
@papakai sure maybe you can join https://gitter.im/gosiris/siris and write me there when you finde time or write me directly at gitter to discuss what should be next etc.

**Comment by KaiSchwarz-cnic** [comment]
@Dexus why gosiris chat room and not PEM ? :)

**Comment by Dexus** [comment]
because i only fast copy one chat link 😄

**Pull request merged**